File: Algoritmo(C)/Alg_S7_7.c
#include <stdio.h>
int main(){
    //variables
    int id, n, quantidade = 0, esfera = 0, limpeza = 0, cabo = 0, quebrado = 0;
    float pesfera = 0, plimpeza = 0, pcabo = 0, pquebrado = 0;
    //input
    printf("Digite o ID do mouse: ");
    scanf("%d",&id);
    while (id != 0){
        printf("[1] Necessita da esfera\n");
        printf("[2] Necessita de limpeza\n");
        printf("[3] Necessita troca do cabo ou conector\n");
        printf("[4] Quebrado ou inutilizado\n");
        scanf("%d",&n);
        //process
        switch (n){
            case 1 :
                esfera = (esfera + 1);
                break;
            case 2 :
                limpeza = (limpeza + 1);
                break;
            case 3 :
                cabo = (cabo + 1);
                break;
            case 4 :
                quebrado = (quebrado + 1);
                break;
            default :
                printf("Numero invalido\n");
        }
        quantidade++;
        printf("Digite o ID do mouse: ");
        scanf("%d",&id);
    }
    pesfera = (((float)esfera * 100) / (float)quantidade);
    plimpeza = (((float)limpeza * 100) / (float)quantidade);
    pcabo = (((float)cabo * 100) / (float)quantidade);
    pquebrado = (((float)quebrado * 100) / (float)quantidade);
    //output
    printf("Situacao \t\t\t\tQuantidade \tPercentual\n");
    printf("1 - Necessita da esfera \t\t%d \t\t%2.f\n",esfera,pesfera);
    printf("2 - Necessita de limpeza \t\t%d \t\t%2.f\n",limpeza,plimpeza);
    printf("3 - Necessita cabo ou conector \t\t%d \t\t%2.f\n",cabo,pcabo);
    printf("4 - Quebrado ou inutilizado \t\t%d \t\t%2.f\n",quebrado,pquebrado);
}

File: Algoritmo(Python)/Alg_S7_7.py
#variables
quantidade = 0
q1 = 0
q2 = 0
q3 = 0
q4 = 0
#input
print("0 - Finaliza")
cod = int(input("Digite o id do mouse: "))
while(cod != 0):
    print("[1] Necessita de esfera")
    print("[2] Necessita de limpeza")
    print("[3] Necessita troca do cabo ou conector")
    print("[4] Quebrado ou inutilizado")
    defeito = int(input("Digite a situação do mouse: "))
    #process
    while((defeito < 1) or (defeito > 4)):
        print("Número inválido")
        defeito = int(input("Digite a situação do mouse: "))
    quantidade += 1
    if(defeito == 1):
        q1 += 1
    elif(defeito == 2):
        q2 += 1
    elif(defeito == 3):
        q3 += 1
    elif(defeito == 4):
        q4 += 1
    p1 = (q1 * 100 / quantidade)
    p2 = (q2 * 100 / quantidade)
    p3 = (q3 * 100 / quantidade)
    p4 = (q4 * 100 / quantidade)
    cod = int(input("Digite o id do mouse: "))
#output
print("Quantidade de mouses: {0}".format(quantidade))
print("SITUAÇÃO                                     QUANTIDADE    PERCENTUAL")
print("[1] Necessita de esfera                         {0}            {1:.2f}".format(q1, p1))
print("[2] Necessita de limpeza                        {0}            {1:.2f}".format(q2, p2))
print("[3] Necessita troca do cabo ou conector         {0}            {1:.2f}".format(q3, p3))
print("[4] Quebrado ou inutilizado                     {0}            {1:.2f}".format(q4, p4))